docs/enhancements/fep-002-config-hooks.md

```diff
@@ -0,0 +1,204 @@
+# EP: Definition of the Fleet and Device API spec
+<!-- this format is inspired by the K8S KEP format https://raw.githubusercontent.com/kubernetes/enhancements/master/keps/NNNN-kep-template/README.md -->
+## Summary
+
+This RFE defines a mechanism for per-device configuration extensiblity,
+where the device configuration can be instructed to talk to an external
+system or hook to fetch specific configurations for the device. This
+RFE also defines a talk-back mechanism for those hooks to tell flightctl
+that such configuration needs to be updated/revoked/etc.
+
+## Motivation
+
+Integration with external systems when devices may need specific
+per-device tokens, per-device configurations sometimes transient
+is a not-well covered use case in our API. Enabling such thing
+with today's API would not be possible. This is why we propose
+this configuration hooks API.
+
+This could be useful i.e. to integrate with ACM, where a service could
+be running on the ACM cluster, and the fleets configured to have
+the devices onboarded to ACM would point to that service in the fleet
+configuration items to fetch the configuration at specific
+points in time (hook points), anyway this is just an example, the
+mechanism should be generic to enable customers to leverage this.
+
+### Goals
+
+* Define hook points in the device configuration or flightctl
```

Comment:
I can see a need to emit fleet/device lifecycle events to an external system (say, to trigger a job via AAP's Event-Driven Automation), but not for the purpose of / being limited to configuration. Would that then duplicate/overlap with the proposed configuration provider's event notifications?

Reply:
Are you referring to another RFE or this?

In this case, I am talking about the configuration entry, which would be the source/reason for such events. Of course, we could model this in a different way altogether.